qiskit/qasm3/ast.py

```diff
@@ -478,6 +478,15 @@ def __init__(self, indexIdentifierList: List[Identifier]):
         self.indexIdentifierList = indexIdentifierList
 
 
+class QuantumGlobalBarrier:
+    """
+    quantumBarrier
+        : 'barrier'
+    """
+
+    __slots__ = ()
```

Comment:
Could this be unified sensibly with `QuantumBarrier`, so we just represent `qargs = []` if it's meant to be the empty list?  It could simplify processing a little.